bee-protocol/src/workers/peer/manager.rs

```diff
@@ -53,33 +63,72 @@ where
         let tangle = node.resource::<Tangle<N::Backend>>();
         let requested_milestones = node.resource::<RequestedMilestones>();
         let metrics = node.resource::<NodeMetrics>();
+        let network_command_tx = node.resource::<NetworkCommandSender>();
+
         let hasher = node.worker::<HasherWorker>().unwrap().tx.clone();
         let message_responder = node.worker::<MessageResponderWorker>().unwrap().tx.clone();
         let milestone_responder = node.worker::<MilestoneResponderWorker>().unwrap().tx.clone();
         let milestone_requester = node.worker::<MilestoneRequesterWorker>().unwrap().tx.clone();
 
+        let PeerManagerConfig {
+            network_rx,
+            peering_rx,
+            network_name,
+        } = config;
+
+        if let Some(peering_rx) = peering_rx {
+            node.spawn::<Self, _, _>(|shutdown| async move {
+                use AutopeeringEvent::*;
+
+                info!("Autopeering handler running.");
+
+                let mut receiver = ShutdownStream::new(shutdown, UnboundedReceiverStream::new(peering_rx));
+
+                while let Some(event) = receiver.next().await {
+                    trace!("Received event {:?}.", event);
+
+                    match event {
+                        IncomingPeering { peer, .. } => {
+                            handle_new_peering(peer, &network_name, &network_command_tx);
+                        }
+                        OutgoingPeering { peer, .. } => {
+                            handle_new_peering(peer, &network_name, &network_command_tx);
+                        }
+                        PeeringDropped { peer_id } => {
+                            handle_peering_dropped(peer_id, &network_command_tx);
+                        }
+                        _ => {}
```

Comment:
shouldn't the non-handled message be logged at least?

Reply:
isn't it? (see line 88)